com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -68,6 +69,24 @@ public class Bob {
     private static File rootFolder = null;
     private static boolean luaInitialized = false;
 
+    public static class OptionValidationException extends Exception {
+        public final int exitCode;
+
+        public OptionValidationException(int exitCode) {
+            this.exitCode = exitCode;
+        }
+    }
+
+    public static class InvocationResult {
+        public final boolean success;
+        public final List<TaskResult> taskResults;
+
+        public InvocationResult(boolean success, List<TaskResult> taskResults) {
+            this.success = success;
+            this.taskResults = taskResults;
+        }
+    }
```

Comment:
The `success` flag is needed for them `main` to pick the right exit code. The `taskResults` field will be used by the editor to create the build errors.